### PR TITLE
Validation fixes

### PR DESCRIPTION
**Pull request by rpavlik**
Gets the validation layer further in actually working with Hello_xr as well as a local headless sample app.

### CODE REVIEW

File: src/api_layers/core_validation.cpp [hunk 1]
```diff
@@ -635,19 +633,34 @@ XrResult CoreValidationXrCreateSession(XrInstance instance, const XrSessionCreat
             }
             cur_ptr = reinterpret_cast<const XrBaseInStructure *>(cur_ptr->next);
         }
-        if (num_graphics_bindings_found != 1) {
+        auto const &enabled_extensions = gen_instance_info->enabled_extensions;
+        bool has_headless = enabled_extensions.end() !=
+                            std::find(enabled_extensions.begin(), enabled_extensions.end(), XR_KHR_HEADLESS_EXTENSION_NAME);
+        bool got_right_graphics_binding_count = num_graphics_bindings_found == 1;
```
Comment (daveh-lunarg) [inline src/api_layers/core_validation.cpp]: Wouldn't mind a set of parentheses to show the precedence explicitly here

File: src/api_layers/core_validation.cpp [hunk 2]
```diff
@@ -635,19 +633,34 @@ XrResult CoreValidationXrCreateSession(XrInstance instance, const XrSessionCreat
...
+        if (!got_right_graphics_binding_count && has_headless) {
+            // This permits 0 as well.
+            got_right_graphics_binding_count = num_graphics_bindings_found == 0;
```
Comment (daveh-lunarg) [inline src/api_layers/core_validation.cpp]: Parens

**Review by daveh-lunarg — APPROVED** [review]
Looks good.

Comments for a slight preference to explicitly show precedence on multiple operator statements.

**Comment by rpavlik** [comment]
Sure, those look good to me, will do.

File: src/api_layers/core_validation.cpp (hunk 1 above)
Comment (rpavlik) [reply src/api_layers/core_validation.cpp]: done! (Also added on the previous line - the lack of a nice `std::contains` function causing that iterator-filled monstrosity has bugged me for a long time...)

**Comment by rpavlik** [comment]
OK, patches revised accordingly.

**Pull request merged**